### DISCUSSION TITLE
一個 PAGFile 可以被多次加入 PAGComposition 嗎?

### DISCUSSION THREAD

**Discussion by newlife70723** [Tencent/libpag#2123] [Q&A] [answered]
一個 PAGFile 可以被多次加入 PAGComposition 嗎? 
我嘗試多次都失敗，我的想像中，PAGFile 被丟入 PAGComposition 中，PAGComposition 會有自己的 PAGLayer 指向該層，我可以單純控制該層的動畫移動，還是 PAGFile 其實跟 PAGComposition 內的 PAGLayer 是同一個物件呢?

**Comment by kevingpqi123** [comment, marked as answer]
这类问题更建议你说明自己想实现的功能，不同的场景答案是不一样的。
一个 PAGFile 当然可以被多次加入 PAGComposition，但这种情况下，由于共用的是同一个指针地址，一旦 PAGFile 的属性被改变，会影响到所有添加的地方，在代码编程中，只有通过 new、copy、alloc 等操作才会产生新的指针地址，如果没有这些操作，共用的都是同一个指针
